cmd/hooks/prestop.go

```diff
@@ -0,0 +1,113 @@
+package hooks
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog/v2"
+)
+
+func PreStop(clientset kubernetes.Interface, timeout time.Duration) error {
```

Comment:
Could we add a doc or file comment about what the point of this hook is, and what the happy & bad cases look like? You described it well in the PR description.

What I would like to see:
- 1-2 sentence description of the 6 min issue (like the one you have in PR description)
- Happy case: This hook attempts to force the node to wait until all volumeAttachments are detached
- Unhappy case: Context times out, customer forced to wait for 6 min force detach.

Reply:
Added relevant context as a comment.

Reply:
For posterity in case my original comment was misunderstood:

If I were not in the meetings that discussed the 6+ minute delays, I would not have the context to understand why this prestop hook should be here. Therefore we should provide a bit of context as a comment, especially if it is added in a clean way as was done in the edit to this PR. 

@torredil the comment you added meets my standards thank you!